doc/architecture/trace_file_formats.adoc

```diff
@@ -16,9 +19,187 @@ The length does not include the integer itself.
+  This meta-data record identifies the file as an OSI multi channel trace file.
+- The file must contain a schema record in the summary section for each top-level message type that is used in one or more OSI channels with the OSI message schema as defined in section <<sec-osi-trace-file-multi-schema-record>>.
+- The file must contain at least one OSI message stream in a channel as defined in section <<sec-osi-trace-file-multi-channel>>.
+- The file may contain additional non-OSI message streams in other channels.
+
+NOTE: All OSI and non-OSI message streams stored in the same file share a common time base.
+Storing of unrelated message streams in one trace file is therefore not generally useful and should be avoided.
+
+[#sec-osi-trace-file-multi-global-meta-data]
+=== Multi channel trace file global meta-data
+
+The file must contain a meta-data record with the `name` field being `net.asam.osi.trace` and the following mandatory `metadata` entries:
+
+`version`::
+The semantic version number of the OSI release that this OSI trace file conforms to.
+This is a string in the format `major.minor.patch`, for example `3.0.0`.
+Note that this version number is not necessarily the same as the OSI schema version(s) used in the trace file.
+Rather, it indicates the version of the OSI trace file format itself.
+
+`min_osi_version`::
+The minimum version of the OSI schema used in the trace file OSI channels.
+This is a string in the format `major.minor.patch`, for example `3.0.0`.
+
+`max_osi_version`::
+The maximum version of the OSI schema used in the trace file OSI channels.
+This is a string in the format `major.minor.patch`, for example `3.0.0`.
+
+`min_protobuf_version`::
+The version of the protobuf implementation used in the trace file.
+This is a string in the format `major.minor.patch`, for example `3.17.3`.
+
+`max_protobuf_version`::
+The version of the protobuf implementation used in the trace file.
+This is a string in the format `major.minor.patch`, for example `3.17.3`.
+
+`zero_time`::
+The point in time corresponding to time 0 in all timestamps in the trace file.
+Must be provided as a full ISO8601 formatted date time string, including timezone data, conforming to the https://www.w3.org/TR/xmlschema11-2/#dateTimeStamp[XML Schema dateTimeStamp] lexical space.
+Values must match the following regular expression:
+`-?([1-9][0-9]{3,}|0[0-9]{3})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])T(([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\.[0-9]+)?|(24:00:00(\.0+)?))(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))`
++
+NOTE: that even in pure simulation use cases there is usually a relationship to real time, as on-board components and environment simulation have necessary relationships to real time (for example, the embedded `HostVehicleData` will carry relevant real time information, for localisation and other purposes).
+
+`creation_time`::
+The point in time when the trace file was created.
```

Comment:
Is this the beginning or end of writing the trace file?

Reply:
I don't know, this is taken from Timm's definitions. Personally I would actually not care very much either way, as without additional recording related meta-data (when, how why, with which equipment was this recorded) I don't see very much use for this field for real measurements, and for simulation use the file system timestamps are usually sufficient.

Reply:
Then I propose to put the end time in there, when the mcap file was finalized.

Reply:
Finalized time stamp would however mean that the meta-data record can only be written at the end of recording (at the end of the data section, as metadata records are not allowed in the summary section at the end). Not impossible to do, but has this effect - start of recording would mean the meta-data record can be written at the beginning (not that this matters all that much, just to keep in mind in discussions).

Reply:
Other tools e.g. ros also use the start of the recording. Nonetheless, I would not name it recording as OSI is mainly focusing on simulations (and also enabling real measurements) where it is simply "writing" but not "recording".

Reply:
I don't have a strong opinion on this. We just need to specify it. We could also put the start of writing to the file as creation_time.